### DISCUSSION TITLE
Why DevExpress CollectionView Feels Like Dessert: A .NET MAUI Fan's Lighthearted Rant

### DISCUSSION THREAD

**Discussion by Kremed** [dotnet/maui#24575] [General]
Hello everyone, let's talk about the CollectionView in .NET MAUI. First off, let me make it clear—I’m a die-hard .NET MAUI fan. I've been on board since day one, and there's no turning back for me.

But, let's be fair—the CollectionView in .NET MAUI is… well, let's just say it's not the greatest. When you load more than 30 items, the user experience is downright painful.

Now, the solution? It exists, and it’s called the CollectionView from DevExpress. It's simply amazing—smooth as silk and solves any issue you throw at it. With 500 items? No problem! You won't see a single hiccup in the smoothness of the UI—it’s that good, like... well, let’s just say it's as satisfying as a great meal.

I fully support the MAUI team and appreciate all their hard work, but I have to ask—can't we just take a look at what DevExpress is doing with their CollectionView and rework ours? I mean, it seems so simple! Or am I missing something here?

That being said, I’m genuinely happy and satisfied with the CollectionView in DevExpress. But, I want to use clean, reliable .NET tools. I don't want to be worried about losing third-party support or end up with an app bloated with 50 extra dependencies.

Finally, sure, as developers, we can fix issues like App Shell's font or the return type in a text field not working properly, but we can’t live without smoothly displaying large sets of data using CollectionView. We use it like we use cutlery when we eat—it’s essential!

**Comment by JRosanowski** [comment]
Thanks for the suggestion.
I had two problems with .Net Maui CollectionView
- It doesn't take up the full view width when rotating on Android
- Tapping an item on iOS would result in all the items disappearing. 

DevExpress CollectionView has neither of these issues. I'd also rather not be dependent on a third party library but have no choice.

**Reply by Kremed** [reply]
I have no issues 

Uploading Screen_Recording_20240916_010602.mp4 …

**Comment by JRosanowski** [comment]
I've actually run into some issues with DxCollectionView as well. Has anyone got any advice about updating the items in a CollectionView on iOS without running into problems like crashes or all the items disappearing, or the UI becoming unresponsive? I seem to have no problems at all on Android but iOS is a nightmare.

**Comment by CodingOctocat** [comment]
DxCollectionView doesn't seem to have an add/remove item animation

**Comment by JRosanowski** [comment]
It looks like some methods of ObservableCollection will cause problems
- Items.Clear() followed by Items.Add all items in a loop.
- Items.Insert(0, item)

What appears to be ok is
- Items.Add(item) followed by Items.Move(Items.Count - 1, 0)
Possibly multiple Adds or multiple Removes are ok although I haven't been able to confirm.
